### PR TITLE
Rename Beam.get_direction

### LINKED ISSUE #6: Beam.get_direction: proposed name change
The dxtbx `Beam` class has a method called `Beam.get_direction`, but unfortunately this invites confusion. Without looking into the details of `Beam`, a developer might reasonably expect this to return a unit vector pointing in the direction of propagation of the beam. In fact, it returns the inverse of that, i.e. the sample-to-source direction (the method `Beam.get_unit_s0` does return what is expected). This trap has claimed at least one victim (https://github.com/cctbx/cctbx_project/pull/133) and needs some care to disentangle.

The imgCIF coordinate system describes this direction using the 'source axis'.
http://www.iucr.org/__data/iucr/cifdic_html/2/cif_img.dic/Caxis.html

>  Axis 3 (Z): The Z-axis is derived from the source axis which goes from
>  the sample to the source.  The Z-axis is the component of the source axis
>  in the direction of the source orthogonal to the X-axis in the plane
>  defined by the X-axis and the source axis.

Perhaps we could come up with a better method name that doesn't conceal an ambiguity. Maybe `Beam.get_unit_source_axis` or `Beam.get_source_direction`?

### PR DESCRIPTION
**Pull request by dagewa**
Beam.get_direction --> Beam.get_sample_to_source_direction.
Fixes #6

Made a PR to make the API change visible, but it turns out to be trivial, with only a handful of changes across DIALS, xia2 and cctbx

### CODE REVIEW

**Comment by dagewa** [comment]
The travis build fails, because it needs https://github.com/cctbx/cctbx_project/tree/beam-get-sample-to-source-direction

File: model/__init__.py [hunk 1]
```diff
@@ -801,3 +801,26 @@ def as_file(self, filename, **kwargs):
...
+        return self.get_sample_to_source_direction()
+
+
+boost.python.inject_into(Beam)(BeamDeprecatedMethods)
+boost.python.inject_into(Crystal)(CrystalAux)
+boost.python.inject_into(Detector)(DetectorAux)
+boost.python.inject_into(Experiment)(ExperimentAux)
+boost.python.inject_into(ExperimentList)(ExperimentListAux)
+boost.python.inject_into(MosaicCrystalKabsch2010)(MosaicCrystalKabsch2010Aux)
+boost.python.inject_into(MosaicCrystalSauter2014)(MosaicCrystalSauter2014Aux)
```
Comment (Anthchirp) [inline model/__init__.py]: these are all obsolete and should go (they're now declared using decorators and the names for injected classes have all gone by now)

File: model/__init__.py [hunk 2]
```diff
@@ -801,3 +801,26 @@ def as_file(self, filename, **kwargs):
         else:
             ext_str = "|".join(j_ext + p_ext)
             raise RuntimeError("expected extension {%s}, got %s" % (ext_str, ext))
+
+
+class BeamDeprecatedMethods(object):
```
Comment (Anthchirp) [inline model/__init__.py]:
```suggestion
class _(object):
```

File: model/__init__.py [hunk 3]
```diff
@@ -801,3 +801,26 @@ def as_file(self, filename, **kwargs):
         else:
             ext_str = "|".join(j_ext + p_ext)
             raise RuntimeError("expected extension {%s}, got %s" % (ext_str, ext))
+
+
```
Comment (Anthchirp) [inline model/__init__.py]: add decorator:
```
@boost.python.inject_into(Beam)
```

File: model/__init__.py [hunk 4]
```diff
@@ -801,3 +801,26 @@ def as_file(self, filename, **kwargs):
...
+class BeamDeprecatedMethods(object):
+    def get_direction(self):
+        import warnings
+
+        warnings.warn(
+            "Calling get_direction is deprecated. Please use "
+            "get_sample_to_source_direction instead. "
```
Comment (Anthchirp) [inline model/__init__.py]:
```suggestion
            ".get_sample_to_source_direction() instead. "
```

File: model/__init__.py (hunk 1 above)
Comment (dagewa) [reply model/__init__.py]: Apologies, they came back from editing the old branch. That explains why merge failed...

**Pull request merged**